x/cdp/types/cdp.go

```diff
@@ -204,29 +185,41 @@ func (augcdps AugmentedCDPs) String() string {
 	return out
 }
 
-// TotalCDPPrincipal is a total principal of a given collateral type
-type TotalCDPPrincipal struct {
-	CollateralType string   `json:"collateral_type" yaml:"collateral_type"` // string representing the unique collateral type of the CDP
-	Amount         sdk.Coin `json:"amount" yaml:"amount"`                   // Amount of principal stored in this CDP
+// TotalPrincipals a collection of TotalPrincipal objects
+type TotalPrincipals []TotalPrincipal
+
+func (t *TotalPrincipal) String() string {
```

Comment:
Do we want to keep implement the stringer methods ourselves? Or use the proto generated ones

Reply:
i vote for the generated stringer methods to reduce code & its usually pretty good i think. Not sure if any downsides.